pkg/kubectl/cmd/config/set_cluster.go

```diff
@@ -169,10 +180,29 @@ func (o createClusterOptions) validate() error {
 		if caPath == "" {
 			return fmt.Errorf("you must specify a --%s to embed", clientcmd.FlagCAFile)
 		}
-		if _, err := ioutil.ReadFile(caPath); err != nil {
-			return fmt.Errorf("could not read %s data from %s: %v", clientcmd.FlagCAFile, caPath, err)
+		fileSize, err := getFileSize(caPath)
+		if err != nil {
+			return err
+		}
+		if fileSize <= 0 {
+			return fmt.Errorf("you must specify a non-empty certificate authority file: %v", caPath)
 		}
 	}
 
 	return nil
 }
+
+// getFileSize attempt open file and get file size.
+// we want to support 'process substitution', so only get stat not read data.
+func getFileSize(filepath string) (int64, error) {
```

Comment:
IMO we don't really need to the file size, we can simply rely on the error returned by `f.Stat()`.

Reply:
i want check file is or isn't empty. so need the file size.